tests/test_check_validation.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+from sys import version_info
+
+from fauxfactory import _check_validation
+
+if version_info[0:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+    if version_info[0] == 2:
+        from mock import Mock
+    else:
+        from unittest.mock import Mock
+
+
+@_check_validation
+def decorated_f():
+    return 'not a number'
+
+
+class CheckValidationTestCase(unittest.TestCase):
+    """_check_validation decorator tests"""
+
```

Comment:
We should add a real case test here:

Example is https://github.com/SatelliteQE/robottelo/issues/4407 which generated string cannot start with number

And also https://github.com/SatelliteQE/robottelo/blob/master/robottelo/libvirt_discovery.py#L21 which gen_mac cannot start with `fe:` 

(then once released we can update those codes to use validators)